fhir-notification-nats/src/main/java/com/ibm/fhir/notifications/nats/impl/FHIRNotificationNATSPublisher.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * (C) Copyright IBM Corp. 2016,2019
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.notifications.nats.impl;
+
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.nats.streaming.AckHandler;
+import io.nats.streaming.Options;
+import io.nats.streaming.StreamingConnection;
+import io.nats.streaming.StreamingConnectionFactory;
+
+import com.ibm.fhir.notification.FHIRNotificationEvent;
+import com.ibm.fhir.notification.FHIRNotificationService;
+import com.ibm.fhir.notification.FHIRNotificationSubscriber;
+import com.ibm.fhir.notification.exception.FHIRNotificationException;
+import com.ibm.fhir.notification.util.FHIRNotificationUtil;
+
+/**
+ * This class implements the FHIR server notification service via a Kafka topic.
+ */
+public class FHIRNotificationNATSPublisher implements FHIRNotificationSubscriber {
+    private static final Logger log = Logger.getLogger(FHIRNotificationNATSPublisher.class.getName());
+    private static FHIRNotificationService service = FHIRNotificationService.getInstance();
+
+    private StreamingConnection sc = null;
+    private AckHandler acb = null;
+    private String channelName = null;
+
+    // "Hide" the default ctor.
+    protected FHIRNotificationNATSPublisher() {
+    }
+
+    public FHIRNotificationNATSPublisher(String clusterId, String channelName, String server) {
+        log.entering(this.getClass().getName(), "ctor");
+        try {
+            init(clusterId, channelName, server);
+        } finally {
+            log.exiting(this.getClass().getName(), "ctor");
+        }
+    }
+
+    /**
+     * Performs any required initialization to allow us to publish events to the topic.
+     */
+    private void init(String clusterId, String channelName, String server) {
+        log.entering(this.getClass().getName(), "init");
+
+        try {
+            this.channelName = channelName;
+            String clientId = "fhir-server";
```

Comment:
best to externalize the variable

Reply:
Agreed.  If we have multiple fhir-servers as nats clients (and we likely would), this would need to be externalized.